extension/chrome/settings/modules/my_key_user_ids.ts

```diff
@@ -29,9 +29,8 @@ View.run(class MyKeyUserIdsView extends View {
     [this.primaryKi] = await KeyStore.get(this.acctEmail, [this.fingerprint]);
     Assert.abortAndRenderErrorIfKeyinfoEmpty(this.primaryKi);
     $('.action_show_public_key').attr('href', this.myKeyUrl);
-    const prv = await PgpKey.read(this.primaryKi.private);
-    const userIds = prv.users.map(u => u.userId).filter(Boolean).map(uid => uid!.userid); // todo - create a common function in settings.js for here and setup.js user_ids
-    Xss.sanitizeRender('.user_ids', userIds.map((uid: string) => `<div>${Xss.escape(uid)}</div>`).join(''));
+    const prv = await KeyUtil.parse(this.primaryKi.private);
+    Xss.sanitizeRender('.user_ids', prv.identities.map((uid: string) => `<div>${Xss.escape(uid)}</div>`).join(''));
```

Comment:
This is interesting - `identities` are the full uid strings, vs `emails` which is parsed / normalized?

Reply:
identities have full uids but only those that have valid email in them. I'll write in a comment

Reply:
Yeah, it seems there are two usages of UIDs: one is for display (`identities`) and one is for lookup (`emails`). This is usually grouped in one property that's an array of objects that have full-UID and e-mail properties inside but based on the usages I saw it seems identities/emails split works well too...